sysl2/sysl/cmd_export.go

```diff
@@ -0,0 +1,58 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/anz-bank/sysl/sysl2/sysl/exporter"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+type exportCmd struct {
+	appName string
+	out     string
+	target  string
+	format  string
+}
+
+func (p *exportCmd) Name() string            { return "export" }
+func (p *exportCmd) RequireSyslModule() bool { return true }
+
+func (p *exportCmd) Configure(app *kingpin.Application) *kingpin.CmdClause {
+	cmd := app.Command(p.Name(), "Export sysl to external types. Supported types: Swagger")
+	cmd.Flag("app-name",
+		"name of the sysl app defined in sysl model."+
+			" if there are multiple apps defined in sysl model,"+
+			" swagger will be generated only for the given app").Required().Short('a').StringVar(&p.appName)
+	cmd.Flag("output", "output filename").Default("output.yaml").Short('o').StringVar(&p.out)
+	cmd.Flag("target", "export target").Default("swagger").Short('t').StringVar(&p.target)
+	cmd.Flag("format", "export format").Default("yaml").Short('f').StringVar(&p.format)
```

Comment:
These 3 flags don't make for a nice user interface. I'd say force the filename to be required (or maybe not, not a huge issue), remove the target flag completly, and change format to be exactly the same as the import flag options, default to "auto" (which means guess from the filename, .yaml == swagger yaml, .json ==swagger .json, etc)

Reply:
Done